pkg/deploy/lattice/listener_manager.go

```diff
@@ -70,16 +72,32 @@ func (d *defaultListenerManager) Upsert(
 	}
```

Comment:
I think we only need to change the comments and not code here right? 

I think we still don't need to call `lattice.UpdateListener()` from any perspective. if the the TG not found (or CREATE_IN_PROGRESS), the earlier code will return the error to prevent calling the `lattice.CreateListener()`.

for TLSRoute backendref svc code path, targetGroupSynthesizer.SynthesizeCreate()  return LATTICE_RETRY first to block to call the `lattice.CreateListener()

 for serviceExport and TLSRoute backendref serviceImport code path, the tgManager.ResolveRuleTgIds() return error to block to call the `lattice.CreateListener()`

Reply:
What would happen here if I 
1. create a TLSRoute with backendRef to serviceA
2. update the TLSRoute replacing backendRef to serviceB

I think with the logic as-is we would not update the listener appropriately. We would find the listener in `findListenerByPort`, then do nothing. Instead, we should be updating the default rule to point to the target group for service B.

Reply:
Ok, got your point, will take a look

Reply:
I did some refactoring in the new revision, please take a look https://github.com/aws/aws-application-networking-k8s/blob/9b3f3234fc91cc74248125a55507b38d338ef623/pkg/deploy/lattice/listener_manager.go#L46-L131

Reply:
Logic-wise, what's here looks good. Resolving.